src/audio/pipeline.c

```diff
@@ -596,7 +599,7 @@ static int pipeline_comp_copy(struct comp_dev *current, void *data, int dir)
 	/* copy to downstream immediately */
 	if (dir == PPL_DIR_DOWNSTREAM) {
 		err = comp_copy(current);
-		if (err < 0)
+		if (err < 0 || err > 0)
```

Comment:
if (err > 0 ) then it's copy stop (which is not an error). So we probably want to change this to 0 ?

Reply:
For other operations like params and prepare it's not modified back to 0, so I wanted to stay aligned. Anyway maybe some higher level function will need to know this status in the future and that it was stopped intentionally.